handler/rfc8628/auth_handler.go

```diff
@@ -55,3 +52,49 @@ func (d *DeviceAuthHandler) HandleDeviceEndpointRequest(ctx context.Context, dar
 	resp.SetInterval(int(d.Config.GetDeviceAuthTokenPollingInterval(ctx).Seconds()))
 	return nil
 }
+
+func (d *DeviceAuthHandler) handleDeviceCode(ctx context.Context, dar fosite.DeviceRequester) (string, error) {
+	code, signature, err := d.Strategy.GenerateDeviceCode(ctx)
+	if err != nil {
+		return "", errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebug(err.Error()))
+	}
+
+	dar.GetSession().SetExpiresAt(fosite.DeviceCode, time.Now().UTC().Add(d.Config.GetDeviceAndUserCodeLifespan(ctx)))
+	if err = d.Storage.CreateDeviceCodeSession(ctx, signature, dar.Sanitize(nil)); err != nil {
+		return "", errorsx.WithStack(fosite.ErrServerError.WithWrap(err).WithDebug(err.Error()))
+	}
+
+	return code, nil
+}
+
+func (d *DeviceAuthHandler) handleUserCode(ctx context.Context, dar fosite.DeviceRequester) (string, error) {
+	userCode, err := retry.DoWithData(
+		func() (string, error) {
+			code, signature, err := d.Strategy.GenerateUserCode(ctx)
+			if err != nil {
+				return "", err
+			}
+
+			dar.GetSession().SetExpiresAt(fosite.UserCode, time.Now().UTC().Add(d.Config.GetDeviceAndUserCodeLifespan(ctx)).Round(time.Second))
+			err = d.Storage.CreateUserCodeSession(ctx, signature, dar.Sanitize(nil))
+			if err != nil {
+				return "", err
+			}
+
+			return code, nil
+		},
+		retry.Attempts(3),
```

Comment:
I would drop the library and simply use standard go

```go
i := 3
for {
   if i < 0 {
      break
   }
   do stuff, if err --> continue aka next loop round
   i := i -1

}
```

reason is we don't need delay, if we are getting a clash due to same code already existent we simply try again straight, I know random generation potentially depends on time but even a ns should be sufficient to change the random string